odhargo/README.md

```diff
@@ -1,37 +1,64 @@
+![Upstream version](https://img.shields.io/github/v/release/argoproj/argo?label=Upstream%20release)
+
+ODH Argo component installs Argo that is namespace bound and not cluster wide. There are two pods running after installation
 
-Argo component installs Argo version 2.7 that is namespace bound and not cluster wide. There are two pods running after installation
 1. Argo Server
 2. Argo Controller
 
 This Argo installation uses the "k8sapi" executor to work on Openshift.
 
 ### Folders
-There is only one folder, `base`. This folder includes the installation yaml files to all necessary Argo resources that need to be installed
 
-### Argo Portal
+1. `cluster` folder contains all the cluster wide resources required to be installed by the operator before Argo can be deployed. It contains all `CustomResourceDefinitions` and `ClusterRoles` which aggregates permissions for those CRDs to admin/edit/view project roles.
 
-This installation creates a route to the Argo portal. To access the portal go to `Routes` and click on the `Argo Portal` route.
+2. There's only a single `base` folder within the `odhargo` folder. This `base` includes all installation yaml files to all necessary namespaced Argo resources.
 
+### Argo Server
+
+This installation creates a route to the Argo portal. To access the portal go to `Routes` and click on the `Argo Server` route.
 
 ### Installation
+
 To install Argo add the following to the `kfctl` yaml file.
 
-```
-  - kustomizeConfig:
-      repoRef:
-        name: manifests
-        path: odhargo
-    name: odh-argo
+```yaml
+- kustomizeConfig:
+    repoRef:
+      name: manifests
+      path: odhargo/cluster
+  name: odhargo-cluster
+- kustomizeConfig:
+    repoRef:
+      name: manifests
+      path: odhargo/odhargo
+  name: odhargo
 ```
 
 ### Run a Workflow
-To run an example workflow, you can use the portal UI to run the example workflow, however the `serviceAccountName` has to be added to the workflow as shown below. We created this example `serviceAccount` as part of the installation to give basic permissions to a workflow. You can edit this `serviceAccount` and add more pemissions if your workflow requires that.
-```
+
+To run an example workflow, you can use the portal UI to submit an [example workflow](odhargo/base/test-workflow.yaml) structured as:
+
+```yaml
 apiVersion: argoproj.io/v1alpha1
 kind: Workflow
 metadata:
   generateName: hello-world-
-spec:
-  serviceAccountName: argo-workflow
+spec: ...
+```
+
+or submit it via [Argo CLI](https://github.com/argoproj/argo/releases):
+
+```sh
+argo submit odhargo/base/test-workflow.yaml
 ```
+
+### Auto-update ODH Argo component manifests
+
+Steps for semi-automated manifests update of Argo manifests in this repo:
+
+1. Execute the [`gen_argo_manifests.sh`](gen_argo_manifests.sh) script within this folder.
+2. Latest manifests are fetched and this `README` is updated.
+3. Commit those changes.
```

Comment:
@tumido no sure why we have a script to auto-update argo? Can you please explain the usecase for this. In our ODH installation we want to specify the release number because we test it.

Reply:
This is targeting the ODH maintainers, because with this script updating the ODH Argo component to be on par with upstream Argo would take a single command. In the PR description you can see what the script does:
1. It fetches the upstream manifests
2. Builds **all** the CRDs into a `cluster/base/crds.yaml` file
3. Copies over **alll** the cluster role aggregations to `cluster/base/cluster-roles.yaml`
4. Builds **all** the resources required for a **namespace install** into a `odhargo/base/namespace-install.yaml`, excluding the CRDs (so we maintain the `cluster` folder = requires cluster-admin permissions and `odhargo` folder = namespaced resources)

This way there's no need for manual copying of files anymore and all is kustomize driven based on the upstream manifests. The result manifests are stored in the same location every time. It changes the current structure of ODH Argo, sure. It makes it less error prone.

Reply:
It will also make the future change sets, when updating Argo to even newer versions, minimal. That's because it's a consistent way to update it. It also follows the upstream [Argo install guide](https://github.com/argoproj/argo/tree/master/manifests) as precisely as possible while maintaining the `odhargo/cluster` and `odhargo/odhargo` apps separation.

Reply:
Moved to https://github.com/opendatahub-io/odh-manifests/pull/289